### PR TITLE
Misc small fixes for lxml

### PR DESCRIPTION
**Pull request by undingen**

### CODE REVIEW

File: src/runtime/util.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ Box* noneIfNull(Box* b) {
     }
 }
 
-Box* coerceUnicodeToStr(Box* unicode) {
+template <ExceptionStyle S> Box* coerceUnicodeToStr(Box* unicode) noexcept(S == CAPI) {
```
Comment (kmod) [inline src/runtime/util.cpp]: I think the raiseExcHelper call in here needs to be changed.

**Comment by kmod** [comment]
lgtm, though it looks like there's a small merge conflict.

File: src/runtime/util.cpp (hunk 1 above)
Comment (undingen) [inline src/runtime/util.cpp]: good catch!

**Comment by undingen** [comment]
I merged this in with a fix for the ```raiseExcHelper``` you mentioned + I added a commit which adds the ```addGCReference``` call for the builtin ```hasattr``` like you have done for ```getattr```

**Pull request merged**